internal/gomod/module.go

```diff
@@ -69,10 +70,14 @@ func (m Module) Hash() (string, error) {
 	return h1, nil
 }
 
-func (m Module) PackageURL() string {
+func (m Module) BOMRef() string {
 	return fmt.Sprintf("pkg:golang/%s?type=module", m.Coordinates())
 }
 
+func (m Module) PackageURL() string {
+	return fmt.Sprintf("pkg:golang/%s?type=module&goos=%s&goarch=%s", m.Coordinates(), os.Getenv("GOOS"), os.Getenv("GOARCH"))
```

Comment:
`GOOS` and `GOARCH` may not always be set explicitly in the environment, so these may end up being empty more often than not.

They *are* however always available via `go env`:

```shell
$ env | rg -i '^go' 
GOROOT=/home/nscur0/.local/go
GOPATH=/home/nscur0/development/go
GO111MODULE=on

$ go env           
GOARCH="amd64"
GOOS="linux"
...

$ GOOS=darwin go env
GOARCH="amd64"
GOOS="darwin"
...
```

We already have a function in `gocmd` that can pull the `go env` data into a `map[string]string`:

https://github.com/CycloneDX/cyclonedx-gomod/blob/fb8866400e5db997c99a12e1294ce306b5d0bf00/internal/gocmd/gocmd.go#L59-L73

We could use global variables for `GOOS` and `GOARCH` in combination with `sync.Once` to avoid redundant shell executions of `go env` every time `PackageURL()` is called.